### PR TITLE
plat-stm32: SCMI reset controllers on remoteproc are disabled on secure fmw loading

### PR DESCRIPTION
**Pull request by etienne-lms**
Changes for STM32 remote proc driver and SCMI server to expose remoteproc reset controllers through SCMI only when the secure loading of remoteproc firmware is disabled. When so, remoteproc reset and power sequence shall go through OP-TEE remoteproc services, not SCMI platform services.

### CODE REVIEW

File: core/arch/arm/plat-stm32mp1/scmi_server.c [hunk 1]
```diff
@@ -570,6 +574,13 @@ int32_t plat_scmi_rd_set_state(unsigned int channel_id, unsigned int scmi_id,
 		return SCMI_DENIED;
 	assert(rd->rstctrl);
 
+#ifdef CFG_STM32MP15
+	/* Remoteproc driver may handle all MCU reset controllers */
+	if (IS_ENABLED(CFG_STM32MP_REMOTEPROC) &
+	    (rd->reset_id == MCU_R || rd->reset_id == MCU_HOLD_BOOT_R))
```
Comment (GseoC) [inline core/arch/arm/plat-stm32mp1/scmi_server.c:578]: ditto

**Comment by GseoC** [comment]
`Reviewed-by: Gatien Chevallier <gatien.chevallier@foss.st.com>`

@arnopo , do you want to give your tag as well?

**Review by arnopo — APPROVED** [review]
LGTM

**Comment by GseoC** [comment]
@etienne-lms I'm just worried about `stm32mp_nsec_can_access_reset()` call above. Who registered the resource?

**Comment by etienne-lms** [comment]
Removal of `stm32mp_nsec_can_access_reset()` and `stm32mp_nsec_can_access_clock()` are addressed in https://github.com/OP-TEE/optee_os/pull/7111.

**Review by GseoC — APPROVED** [review]

**Pull request merged**